### PR TITLE
(PDB-2814) Add class for installing CLI

### PR DESCRIPTION
**Pull request by ajroetker**
This commit adds a class for installing and managing the PuppetDB CLI to
the PuppetDB module.

### CODE REVIEW

**Comment by ajroetker** [comment]
@kbarber @wkalt @senior @rbrw
I'm not sure what exactly this class should look like so I'm putting the PR up for review for now to start discussion.

With the current PR, a user would install the CLI like so

```
include puppetdb::cli
# or
# class { 'puppetdb::cli' : puppetdb_urls => ["https://#{puppetdb_host}:#{puppetdb_port}"] }
```

I have the defaults for the SSL credentials set to the agent certs, but the url set to http and localhost, should I default the SSL credentials at all if assuming the CLI will be installed on the PDB node by default?

Should we be defaulting to localhost for the puppetdb url? We could try and default to using SSL and checking if the puppetdb_host and puppetdb_port params are defined on the puppetdb class.

Should we attempt to validate version compatibility (i.e. <= PDB 3 don't support the CLI)?

If we do try to default to SSL by default for the CLI, should we attempt to warn users that their agent certs (or which certs we default to) need to be on the PuppetDB cert whitelist?

What I think makes the most sense...

Default to nothing for SSL credentials at all and use localhost for the urls by default, and document more advanced usage from there.

**Comment by ajroetker** [comment]
We should wait to merge this until we've released the new version of the CLI.

File: manifests/cli.pp [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  }
+
+  file { $puppetdb_cli_config:
+    ensure  => 'present',
+    owner   => 'root',
+    group   => 'root',
+    mode    => '0444',
+    content => inline_template('<%= @_puppetdb_cli_config.to_json %>'),
+    require => Package['puppet-client-tools'],
+  }
```
Comment (ajroetker) [inline manifests/cli.pp]: All this code is mostly lifted from the PE module (puppetlabs-puppet_enterprise).

File: README.md [hunk 2]
```diff
@@ -856,6 +856,49 @@ be installed from the regular repository. Defaults to `true`.
...
+
+**Parameters within `puppetdb::cli`:**
+
+####`version`
+
+The version of the `puppet-client-tools` package that should be installed. You
+may specify an explicit version number, 'present', or 'latest' (defaults to
+'present').
+
+####`puppetdb_urls`
```
Comment (wkalt) [inline README.md]: why invent a new term here? is it going to conflict with server_urls somewhere else?

Comment (ajroetker) [inline README.md]: I lifted this CLI management from the PE module. We don't actually use `server_urls` anywhere in the module (i.e. we assume one PuppetDB host/port for everything else in the module.

**Comment by ajroetker** [comment]
@wkalt I updated this finally for your comments.

**Comment by wkalt** [comment]
@senior @ajroetker I'm thinking this may be blocked on CT-18. We don't have client tools packages for debian yet, and I've got reservations about introducing module features that are only available for certain supported operating systems.

**Comment by senior** [comment]
@ajroetker sounded like we were waiting on a CLI release or something here? any update?

**Comment by preisbeck** [comment]
Is there any chance that version 5.2.0 will hit the forge soon, or is there a patch released planed? I'd like to use some new parameter that are exclusive to the Git version.

**Pull request closed without merging**